### PR TITLE
fix iOS XML and subscribe bugs, search filter header bar style

### PR DESCRIPTION
**Pull request by toddmedema**
Closes https://github.com/ExpeditionRPG/expedition-app/issues/320

Moves the app to Cheerio!

### CODE REVIEW

File: app/actions/Web.tsx [hunk 1]
```diff
@@ -4,31 +4,28 @@ import {toCard} from './Card'
...
+const Cheerio = require('cheerio'); // Doesn't like import statements
 
 export function fetchQuestXML(id: string, url: string) {
   return (dispatch: Redux.Dispatch<any>): any => {
-    $.get(url, function(data: XMLElement | string) {
-      dispatch(loadQuestXML(id, data, defaultQuestContext()));
+    $.ajax({url,
+      dataType: 'text',
+      success: (data: string) => {
+        const quest = Cheerio.load(data)('quest');
```
Comment (smartin015) [inline app/actions/Web.tsx]: If the XML is malformed, this could throw an exception. Have we tested against this?

File: app/actions/Web.tsx [hunk 2]
```diff
@@ -96,7 +93,7 @@ export function subscribe(email: string) {
       if (xhr.status === 200) {
         window.FirebasePlugin.logEvent('user_subscribe', email);
       } else {
-        window.FirebasePlugin.logEvent('user_subscribe_err', err);
+        // window.FirebasePlugin.logEvent('user_subscribe_err', xhr.status); // causes crash on iOS, also not critical
```
Comment (smartin015) [inline app/actions/Web.tsx]: p'haps we can just remove this line for now.

File: app/quests/test_quest.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <choice text="End now">
       <trigger data-line="19">end</trigger>
     </choice>
-    <choice text="Fight">
+    <choice text="Fight one={{one}}">
```
Comment (smartin015) [inline app/quests/test_quest.xml]: What is `one`?

Comment (toddmedema) [inline app/quests/test_quest.xml]: It's equal to 1 - it's just there as a visual check that variable rendering is working (it is the test quest, after all :)

File: app/actions/Web.tsx (hunk 2 above)
Comment (toddmedema) [inline app/actions/Web.tsx]: Fair. Been thinking about moving off Firebase / to Google Analytics anyway

File: app/actions/Web.tsx (hunk 1 above)
Comment (toddmedema) [inline app/actions/Web.tsx]: Good idea. Just tried:
```
const foo = Cheerio.load('fsfidjsoj<fd></sf');
console.log(foo);
```
And no errors were thrown / foo was a valid Cheerio object. Seems like we're good against malformed XML, then (valid QDL XML is a separate issue....)

Comment (smartin015) [inline app/actions/Web.tsx]: It'd be nice to have this be a unit test, e.g. fake web response with bogus XML, to ensure we surface some error. Split this into #331.

**Pull request merged**